pkg/rotator/rotator.go

```diff
@@ -45,14 +45,17 @@ const (
 
 var crLog = logf.Log.WithName("cert-rotation")
 
-var vwhGVK = schema.GroupVersionKind{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "ValidatingWebhookConfiguration"}
-
 var crdGVK = schema.GroupVersionKind{Group: "apiextensions.k8s.io", Version: "v1beta1", Kind: "CustomResourceDefinition"}
 
```

Comment:
Is there any way we can generalize this to cover the CRD webhooks as well?

Reply:
Thought about this one too, but this is a bit tricky due a different schema in a regular webhook and a conversion webhook.
In validating/mutating webhook we modify:
webhooks:             <---- array, we modify for each element
  - clientConfig: caBundle: <VALUE>
while in a conversion webhook, we modify the crd path:
spec: conversion: webhookClientConfig: caBundle: <VALUE>

I changed it, with having to state which type of webhook is used (not so nice, but the simplest).
Alternatively we could either check the requested GVK inside injectCert() (also not so nice),
or pass the modification path as a parameter (nice, but an overkill to me).

Reply:
I'm not 100% sure I understand what you're saying? I'm mainly talking about generalizing as much of the _interface_ as possible, I understand that some of the _implementation_ might be very different between CRD webhooks and VWC/MWC webhooks.

Reply:
Just wanted to note that there are some differences and we need to pass the type of the webhook along the name.